### PR TITLE
✨ Enable to export followers list 

### PR DESCRIPTION
**Pull request by AuroraHuang22**
截圖 2023-03-07 下午4 45 41
截圖 2023-03-07 下午4 45 27

https://user-images.githubusercontent.com/75730405/223371271-d00bec2e-0e9d-4f1c-811e-46bd726701d5.mp4

### CODE REVIEW

File: src/mixins/wallet.js [hunk 1]
```diff
@@ -93,5 +106,48 @@ export default {
...
     },
+    async fetchFollowers() {
+      try {
+        await this.walletFetchFollowers();
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.error(error);
+      }
+    },
+    exportFollowerList() {
```
Comment (nwingt) [inline src/mixins/wallet.js:117]:
```suggestion
    async exportFollowerList() {
```

File: src/mixins/wallet.js [hunk 2]
```diff
@@ -93,5 +106,48 @@ export default {
...
+          if (err) {
+            // eslint-disable-next-line no-console
+            console.error(err);
+            reject(err);
+          }
+
+          const csvBlob = new Blob([csvString], {
+            type: 'text/csv;charset=utf-8;',
+          });
+          // // Download CSV file
```
Comment (nwingt) [inline src/mixins/wallet.js:140]:
```suggestion
          // Download CSV file
```

File: src/mixins/wallet.js (hunk 1 above)
Comment (AuroraHuang22) [reply src/mixins/wallet.js:117]: This function doesn't use await and only returns a Promise, so do we still need to be declared as async？

File: src/locales/zh-Hant.json [hunk 3]
```diff
@@ -391,6 +391,12 @@
   "portfolio_unsubscribe_message_success": "已取消追蹤 {creator}。",
   "portfolio_unsubscribe_success_alert": "已取消追蹤 {creator}",
   "portfolio_unsubscribe_title": "取消追蹤 {creator}",
+  "portfolio_follower_title": "誰在追蹤我?",
```
Comment (williamchong) [inline src/locales/zh-Hant.json:394]:
```suggestion
  "portfolio_follower_title": "我的追蹤者",

```

File: src/mixins/wallet.js (hunk 1 above)
Comment (williamchong) [reply src/mixins/wallet.js:117]: How about just use sync mode
https://csv.js.org/stringify/api/sync/

File: src/store/modules/wallet.js [hunk 4]
```diff
@@ -523,6 +537,24 @@ const actions = {
...
+    try {
+      if (state.isFetchingFollowers) return;
+      commit(WALLET_SET_FOLLOWERS_FETCHING_STATE, true);
+      const { followers } = await this.$axios.$get(getUserV2Followers());
+      commit(WALLET_SET_FOLLOWERS, followers);
+      if (followers.length) {
+        dispatch('lazyGetUserInfoByAddresses', followers);
+      }
+    } catch (error) {
+      throw error;
```
Comment (williamchong) [inline src/store/modules/wallet.js:553]: Skip the catch block if we are not catching it

File: src/components/FollowerDialog.vue [hunk 5]
```diff
@@ -0,0 +1,86 @@
+<template>
+  <Dialog
+    :open="isOpenFollowersDialog"
```
Comment (williamchong) [inline src/components/FollowerDialog.vue:3]: Just curious, why not use `v-if` on the whole component, but a `isOpenFollowersDialog` is used on Dialog open?

File: src/components/FollowerDialog.vue [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+          'transition',
+          'duration-75',
+          'ease-in',
+          'hover:bg-light-gray',
+        ]"
+      >
+        <NuxtLink
+          class="flex items-center text-like-green"
+          :to="`/${follower.wallet}`"
```
Comment (williamchong) [inline src/components/FollowerDialog.vue:38]: Use named route

File: src/components/FollowerDialog.vue [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+          target="_blank"
+        >
+          <Identity
+            :avatar-url="follower.avatar"
+            :avatar-size="32"
+            :is-lazy-loaded="true"
```
Comment (williamchong) [inline src/components/FollowerDialog.vue:44]: Do we (need to) know if its civic liker?

File: src/mixins/wallet.js [hunk 8]
```diff
@@ -1,4 +1,6 @@
 import { mapActions, mapGetters } from 'vuex';
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (williamchong) [inline src/mixins/wallet.js:2]: Shout not disable

File: src/mixins/wallet.js [hunk 9]
```diff
@@ -93,5 +106,48 @@ export default {
         'menubar=no,location=no,width=576,height=768'
       );
     },
+    async fetchFollowers() {
+      try {
+        await this.walletFetchFollowers();
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.error(error);
+      }
```
Comment (williamchong) [inline src/mixins/wallet.js:115]: `fetchFollowers` is not a good name for a function that purpose is wrapping `walletFetchFollowers` with try catch
`fetchFollowersWithTryCatch` maybe, idk @nwingt any comment?

File: src/pages/_id/index.vue [hunk 10]
```diff
@@ -61,6 +61,27 @@
...
+          :is-empty="false"
+        />
+
+        <!-- Follower List -->
+        <div v-if="walletHasLoggedIn" class="flex items-center justify-center">
+          <div
+            class="underline transition-all duration-75 cursor-pointer text-medium-gray hover:text-dark-gray"
+            @click="handleClickFollowers"
+          >{{ $t('portfolio_follower_title') }}</div>
+        </div>
```
Comment (williamchong) [inline src/pages/_id/index.vue:82]: Is this only shown if user is the owner of this porfolio?

**Review by williamchong — CHANGES_REQUESTED** [review]

File: src/components/FollowerDialog.vue (hunk 5 above)
Comment (nwingt) [reply src/components/FollowerDialog.vue:3]: Use `open` instead of `v-if` to provide a transition on opening

File: src/mixins/wallet.js (hunk 9 above)
Comment (nwingt) [reply src/mixins/wallet.js:115]: I think we should called walletFetchFollowers directly if we there is no other action
Also, I don't think we should put `walletFetchFollowers` in wallet mixin, as this action is not commonly used

File: src/pages/dashboard/index.vue [hunk 11]
```diff
@@ -272,7 +272,12 @@ export default {
         1
       );
       this.isOpenFollowersDialog = true;
-      await this.fetchFollowers();
+      try {
+        await this.walletFetchFollowers();
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.error(error);
+      }
```
Comment (williamchong) [inline src/pages/dashboard/index.vue:280]: In this same commit, `walletFetchFollowers` has a huge try catch block that catches any axios error
Do we really need this try catch?

**Review by williamchong — APPROVED** [review]

File: functions/package.json [hunk 12]
```diff
@@ -39,6 +39,7 @@
     "cookie-parser": "^1.4.4",
     "cosmjs-types": "^0.5.0",
     "create-hash": "^1.2.0",
+    "csv-stringify": "^6.3.0",
```
Comment (nwingt) [inline functions/package.json:42]: Should remove this if not used

**Review by nwingt — COMMENTED** [review]
Can undo the changes in `package-json.lock`?

**Comment by nwingt** [comment]
@AuroraHuang22 Please resolve conflict

File: src/mixins/wallet.js [hunk 13]
```diff
@@ -96,5 +108,37 @@ export default {
...
+      // Download CSV file
+      const csvUrl = URL.createObjectURL(csvBlob);
+      const hiddenLink = document.createElement('a');
+      hiddenLink.href = csvUrl;
+      hiddenLink.target = '_blank';
+      hiddenLink.download = 'my-followers.csv';
+      hiddenLink.click();
+    },
+
+    escapeCsvField(field) {
```
Comment (nwingt) [inline src/mixins/wallet.js:137]: Should move to util?
```suggestion
    escapeCSVField(field) {
```

**Review by williamchong — APPROVED** [review]

**Pull request merged**